src/shogun/mathematics/graph/Tensor.h

```diff
@@ -197,15 +135,23 @@ namespace shogun
 
 #endif
 
+		private:
 			[[nodiscard]] size_t get_size_from_shape(const Shape& size) const {
 				return std::accumulate(
 				    size.begin(), size.end(), size_t{1}, std::multiplies{});
 			}
 
-			private :
-			    // whether the memory is owned by the tensor, i.e. should it be
-			    // deleted by the destructor
-			    bool m_free;
+		public:
+
+			Tensor(const Shape& shape, const std::shared_ptr<NumberType>& type)
+			    : m_free(true), m_data(nullptr), m_shape(shape), m_type(type)
+			{
+				m_data = sg_aligned_malloc(size_in_bytes(), alignment::container_alignment);
+			}
```

Comment:
this should be protected but for some reason it causes issues when calling this constructor from ShogunStorage, even with a friend declaration